MaxText/max_utils.py

```diff
@@ -525,7 +528,7 @@ def create_device_mesh(config, devices=None):
   if devices is None:
     devices = jax.devices()
   num_devices = len(devices)
-  num_slices = config.num_slices
+  num_slices = 1 if config.inference_benchmark_test else config.num_slices
```

Comment:
We will add multi-slice support for inference soon (~Q1) and this code will be confusing when one runs inference benchmark on that setup :). 
 Is this mainly to reduce the time take for the benchmark? Or is this causing some issues with the test?
 
 We also have Jetstream specific config for inference, you can set `num_slices = 1` here and use it for testing -- https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/configs/inference_jetstream.yml

Reply:
afaik, the `num_slices ` get overwritten as the number of the slices at some point in the code, even thought we pass `1` from the config file.

Reply:
Yes num_slices is meant to be automatically determined by jax.devices except in the AOT case

config comment: https://github.com/AI-Hypercomputer/maxtext/blob/4651cb3c73de92e528a98281be29c05b27aaf501/MaxText/configs/base.yml#L296-L299
pyconfig overwriting num_slices: https://github.com/AI-Hypercomputer/maxtext/blob/4651cb3c73de92e528a98281be29c05b27aaf501/MaxText/pyconfig.py#L428
jax.devices logic in max_utils setting num_slices: https://github.com/AI-Hypercomputer/maxtext/blob/4651cb3c73de92e528a98281be29c05b27aaf501/MaxText/max_utils.py#L359